terraform/variables.tf

```
@@ -0,0 +1,60 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+variable "app_name" {
+  description = "Name of the application in the Juju model."
+  type        = string
+  default     = "openstack-integrator"
+}
+
+variable "base" {
+  description = "Ubuntu bases to deploy the charm onto"
+  type        = string
+  default     = "ubuntu@22.04"
+
+  validation {
+    condition     = contains(["ubuntu@20.04", "ubuntu@22.04"], var.base)
+    error_message = "Base must be one of ubuntu@20.04, ubuntu@22.04"
+  }
+}
+
+variable "channel" {
+  description = "The channel to use when deploying a charm."
+  type        = string
+  default     = "1.31/stable"
+}
+
+variable "config" {
+  description = "Application config. Details about available options can be found at https://charmhub.io/openstack-integrator/configurations."
+  type        = map(string)
+  default     = {}
+}
+
+variable "constraints" {
+  description = "Juju constraints to apply for this application."
+  type        = string
+  default     = "arch=amd64"
+}
+
+variable "model" {
+  description = "Reference to a `juju_model`."
```

```suggestion
  description = "Reference to a `juju_model` that this application gets deployed onto."
```
This is a little vague.

---

Cool, i'll improve that.  thanks